base_delivery_carrier_label/models/stock_picking.py

```diff
@@ -116,6 +116,13 @@ def _set_a_default_package(self):
                 package = self.env['stock.quant.package'].create({})
                 move_lines.write({'result_package_id': package.id})
 
+    def _set_carrier_tracking_ref(self, shipping_labels):
+        if len(shipping_labels) == 1:
+            label = shipping_labels[0]
+            self.write(
+                {"carrier_tracking_ref": label.get("tracking_number")}
+            )
```

Comment:
the code looks like there could be more than 1 shipping label per picking and that in that case we will not get a carrier_tracking_ref. 

Should the test be changed to `len(shipping_labels) >= 1`?

Reply:
@bealdav do you have an opinion on this?

Reply:
For this PR I propose we keep the current behavior of both modules and we propose an other change if we want to redesign it.